src/plugins/kdb/lmdb/kdb_lmdb.c

```diff
@@ -737,13 +745,87 @@ klmdb_get_principal(krb5_context context, krb5_const_principal searchfor,
     if (ret)
         goto cleanup;
 
+    ret = kdb5int_extract_db_args_from_tl_data(context, &(*entry_out)->tl_data,
+                                                &(*entry_out)->n_tl_data, &db_args);
+    if ((ret == 0) && (db_args != NULL)) {
+        for (i = 0; db_args[i] != NULL; i++) {
+            if (strncmp(db_args[i], ALIAS_PREFIX, ALIAS_PREFIX_LEN) == 0) {
+                ret = krb5_parse_name(context, db_args[i] + ALIAS_PREFIX_LEN, &alias_princ);
+                if (ret != 0) {
+                    krb5_db_free_principal(context, *entry_out);
+                    *entry_out = NULL;
+                    goto cleanup;
+                }
+                ret = klmdb_get_principal(context, alias_princ, flags, &temp_entry);
+                krb5_free_principal(context, alias_princ);
+                if (ret != 0) {
+                    krb5_db_free_principal(context, *entry_out);
+                    *entry_out = NULL;
+                    k5_setmsg(context, KRB5_KDB_NOENTRY,
+                            _("Alias \"%s\" not found in the database"),
+                            db_args[i] + ALIAS_PREFIX_LEN);
+                    ret = KRB5_KDB_NOENTRY;
+                    goto cleanup;
+                }
+                /* replace entry->princ by the aliasing entry princ */
+                krb5_free_principal(context, temp_entry->princ);
+                krb5_copy_principal(context, (*entry_out)->princ, &temp_entry->princ);
```

Comment:
Why this?  The behavior documented in kdb.h is that the module should return an entry with the canonical name, not the requested name.

Reply:
My mistake, I'll fix it.